src/domain/lottoResultCalculator.js

```diff
@@ -0,0 +1,44 @@
+import { NO_MATCH_PLACE, PERCENTATION } from '../constants/prize-constants.js';
+import excludeKeyFromObject from '../utils/excludeKeyFromObject.js';
+import roundToSecondDecimalPlace from '../utils/roundToSecondDecimalPlace.js';
+
+import prize from './prize.js';
+
+class LottoResultCalculator {
+  #lottoList;
+
+  #winningLotto;
+
+  constructor({ lottoList, winningLottoNumbers, bonusNumber }) {
+    this.#lottoList = lottoList;
+    this.#winningLotto = { winningLottoNumbers, bonusNumber };
+  }
+
+  getTotalResult() {
+    const initialResult = prize.generateInitiallResultObject();
+    const totalResult = this.#lottoList.reduce((acc, lotto) => {
+      const rank = lotto.getRank(this.#winningLotto);
+      if (rank !== NO_MATCH_PLACE) {
+        initialResult[rank] += 1;
+      }
+      return acc;
+    }, initialResult);
```

Comment:
여기 함수를 읽어보면 복잡하게 되어있는데 왜 그럴지 자세히 살펴보니  acc 인자가 실제로 사용되지 않고 있네요
```suggestion
  this.#lottoList.forEach(lotto => {
    const rank = lotto.getRank(this.#winningLotto);
    if (rank !== NO_MATCH_PLACE) {
      initialResult[rank] += 1;
    }
  });
```
이런식으로 바꿔보면 어떨까요?

Reply:
훨씬 간단하고 가독성이 더 좋네요 !! 
저 코드를 짤 당시에 객체에 값을 누적해야겠다는 생각에 저렇게 했던 것 같습니다 ..
수정했습니다 감사합니다